src/testUtils/mocks/dataSources.ts

```diff
@@ -1614,6 +1615,24 @@ export const createMockEcdsaSignature = (
   return createMockStringCodec(signature) as MockCodec<EcdsaSignature>;
 };
 
+/**
+ * @hidden
+ */
+export const createMockBTreeSet = <T extends Codec>(
+  items: BTreeSet<T> | unknown[]
+): MockCodec<BTreeSet<T>> => {
+  if (isCodec<BTreeSet>(items)) {
+    return items as MockCodec<BTreeSet<T>>;
+  }
+  const res = createMockCodec(items, !items) as unknown as BTreeSet;
+  res.has = sinon.stub().returns(false);
+  (res as any).size = items.length;
```

Comment:
Why is this assertion necessary?

Reply:
I think size is a special readonly property of a BTreeSet, so I used this to set it anyway. I forget the exact reason but will look again in the morning